### ISSUE TITLE
[New Feature Request] Meta-attack

### ISSUE THREAD

**Issue by jim-smith** [AI-SDC/SACRO-ML#428]
**Is your feature request related to a problem? Please describe.**
Currently each individual attack can calculate and output the vulnerability of each member of the training set into results.json.

However we typically report measures (accuracy, true positive rate st some fixed false positive rate, PDIF/FDIF rates)  that aggregate over all the records.

What that doesn't tell us is whether some training records are vulnerable to *all* (or most of ) the attacks.
- or to repeated runs of the same attack (since some are stochastic)
- Or whether some test records are always misclassified.

 
**Describe the solution you'd like**
Something that pulled all the relevant individual vulnerabilities into a pandas data frame - one column per attack, one row per train (and possibly test) record.
- that would give us the ability to produce Venn diagrams.
- or other create a 'meta-attack' that scored 'how many different attacks is this record vulnerable to' or (since the vulnerabilities may be attack confidence values in $[0-1]$ - either the geometric or arithmetic mean of those vulnerabilities.

**Describe alternatives you've considered**

**Additional context**
Start by running one of the simple sklearn examples and when you run lira and structural attacks set  `report_individual=True` in the constructor
that should let you see where  the record-level vulnerabilities gt put in results .json

**Comment by jim-smith** [comment]
@JessUWE this the meta attack we talked about about in the sacro-ml meeting and then later when you and I met to discuss tasks over the next 2-3 weeks

**Comment by shamykyzer** [comment]
**Overview**
<img width="600" alt="meta_attack_overview" src="https://github.com/user-attachments/assets/10fda6da-23dd-41ca-b631-926772291c31" />

**Architecture flow with behaviour modes**
<img width="600" alt="meta_attack_architecture" src="https://github.com/user-attachments/assets/51e9a9a7-b36c-479b-80d0-5c6536539947" />

**Aggregation pipeline**
<img width="500" alt="meta_attack_aggregation" src="https://github.com/user-attachments/assets/afc09d34-d718-45e0-9d17-e5f98c16dfad" />

<details>
<summary><b>Class hierarchy</b> (click to expand)</summary>

<img width="400" alt="meta_attack_class_hierarchy" src="https://github.com/user-attachments/assets/3567eebd-c2c5-417f-ba80-cbf8d6c79c92" />

</details>
